components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -177,26 +177,36 @@ SUBROUTINE p3_init_a(lookup_file_dir,version_p3)
     !PMC: deleted ice-ice collision lookup table here b/c only used for nCat>1.
     ! So there is no need to fill lookup values for lookup table 2.
 
+   return
+
   END SUBROUTINE p3_init_a
 
   subroutine p3_get_tables(mu_r_user, revap_user, vn_user, vm_user)
     ! This can be called instead of p3_init_b.
+    implicit none
     real(rtype), dimension(150), intent(out) :: mu_r_user
     real(rtype), dimension(300,10), intent(out) :: vn_user, vm_user, revap_user
     mu_r_user(:) = mu_r_table(:)
     revap_user(:,:) = revap_table(:,:)
     vn_user(:,:) = vn_table(:,:)
     vm_user(:,:) = vm_table(:,:)
+
+   return
+
   end subroutine p3_get_tables
 
   subroutine p3_set_tables(mu_r_user, revap_user, vn_user, vm_user)
     ! This can be called instead of p3_init_b.
+    implicit none
     real(rtype), dimension(150), intent(in) :: mu_r_user
     real(rtype), dimension(300,10), intent(in) :: vn_user, vm_user, revap_user
     mu_r_table(:) = mu_r_user(:)
     revap_table(:,:) = revap_user(:,:)
     vn_table(:,:) = vn_user(:,:)
     vm_table(:,:) = vm_user(:,:)
+
+   return
```

Comment:
I don't mind these, but fwiw, it's nice to reserve `return` for non-default return cases. `end` at the subroutine level has an implicit `return` in Fortran.